src/frontend/src/tests/lib/canisters/reward.canister.spec.ts

```diff
@@ -0,0 +1,151 @@
+import type {
+	NewVipRewardResponse,
+	_SERVICE as RewardService,
+	UserData
+} from '$declarations/rewards/rewards.did';
+import { RewardCanister } from '$lib/canisters/reward.canister';
+import type { CreateCanisterOptions } from '$lib/types/canister';
+import { mockIdentity } from '$tests/mocks/identity.mock';
+import { HttpAgent, type ActorSubclass } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+import { mock } from 'vitest-mock-extended';
+
+vi.mock(import('$lib/actors/agents.ic'), async (importOriginal) => {
+	const actual = await importOriginal();
+	return {
+		...actual,
+		// eslint-disable-next-line require-await
+		getAgent: async () => mock<HttpAgent>()
+	};
+});
+
+describe('reward.canister', () => {
+	const createRewardCanister = ({
+		serviceOverride
+	}: Pick<CreateCanisterOptions<RewardService>, 'serviceOverride'>): Promise<RewardCanister> =>
+		RewardCanister.create({
+			canisterId: Principal.fromText('tdxud-2yaaa-aaaad-aadiq-cai'),
+			identity: mockIdentity,
+			certifiedServiceOverride: serviceOverride,
+			serviceOverride
+		});
+	const service = mock<ActorSubclass<RewardService>>();
+	const mockResponseError = new Error('Test response error');
+	const queryParams = {
+		certified: false
+	};
+
+	describe('getUserInfo', () => {
+		it('returns true if user is vip', async () => {
+			const mockedUserData: UserData = {
+				is_vip: [true],
+				airdrops: [],
+				sprinkles: []
+			};
+			service.user_info.mockResolvedValue(mockedUserData);
+
+			const { getUserInfo } = await createRewardCanister({
+				serviceOverride: service
+			});
+
+			const userData = await getUserInfo(queryParams);
+			expect(service.user_info).toHaveBeenCalledWith();
+			expect(userData.is_vip.length).toBe(1);
+			expect(userData.is_vip[0]!).toBeTruthy();
```

Comment:
Please avoid the use of `!` 

cc we need an eslint rule to prevent this bad practice @AntonioVentilii , thought we already had one

Reply:
Unrelated, you can also use the `fromNullable` utility to decode variant.

Your assertion can look like following for example:

```
expect(fromNullable(userData.is_vip) === true).toBeTruthy();

// or

const isVip = romNullable(userData.is_vip);
expect(isVip ?? false).toBeTruthy();
```